### ISSUE TITLE
Multiple nexts in delegation reporting

### ISSUE THREAD

**Issue by paweljakubas** [cardano-foundation/cardano-wallet#1342]
# Context

U/S

As a wallet client,
I want to reliably know how my delegation changes irrespective of how often certificate is discovered,
So that I can better inform users about their current and next delegation settings.

A/C

The current implementation handles well cases when the certificate is discovered at most once per 2 epochs. We may want to handle situations where : 

In a given epoch there were more than one certificate discovered
Certificate(s) could be discovered in the adjacent epochs
If so, we may want to take the latest certificate in a given epoch to determine what is next. Moreover, if two (or more) certificates are discovered in adjacent epochs we may want to have two `next` in reporting. 

# Decision

We will define WalletDelegation that accommodates list of nexts

# Acceptance Criteria

1. Logic is in Primitive.Types' WalletDelegation
2. Integration test showing the case is added. 

---

# Development 

- Revise api quit endpoint to use a `*` instead of an explicit pool id ==> TODO

# QA

**Comment by piotr-iohk** [comment]
A "corner case":

1. have a wallet delegating to pool A
2. in epoch X delegate to pool B
```
I have:
delegation active = A
delegation next = Target B, changes_at: X+2
```
  3. in epoch X+1 quit pool A
```
I have:
delegation active = A
delegation next1 = Target B, changes_at: X+2
delegation next2= not_delegating, changes_at: X+3
 - a bit wierd, because delegating to Target B, will already automatically quit my active target A, so why my delegation next2 is not_delegating?
```
  4. in X + 2
```
delegation active = B
delegation next = not_delegating, changes_at: X+3
...
```
 5. in X + 3
delegation active = not_delegating
```
So, I have made a request to quit A and as a result I have quit B...
```


## Expected behavior

Perhaps, when you are delegating to A and have already any next delegation target active, you shouldn't be able to request quit A in a first place...

or

```
DELETE /stake-pools/{stakePoolId}/wallets/{walletId}
```
Should not be taking {stakePoolId} as a parameter not to mislead.

**Comment by piotr-iohk** [comment]
I will move to in progress for now until #1386 is done.

**Comment by piotr-iohk** [comment]
LGTM. Testing addressed in #1386.
